src/platform_impl/android/mod.rs

```diff
@@ -747,9 +753,35 @@ impl DeviceId {
 #[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
 pub struct PlatformSpecificWindowAttributes;
 
+fn show_hide_keyboard(app: AndroidApp, show: bool) {
+    use jni::{objects::JObject, JavaVM};
+    let vm = unsafe { JavaVM::from_raw(app.vm_as_ptr() as _).unwrap() };
+    let activity = unsafe { JObject::from_raw(app.activity_as_ptr() as _) };
+    let mut env = vm.attach_current_thread().unwrap();
+    let window = env
+        .call_method(&activity, "getWindow", "()Landroid/view/Window;", &[])
+        .unwrap()
+        .l()
+        .unwrap();
+    let wic = env
+        .call_method(window, "getInsetsController", "()Landroid/view/WindowInsetsController;", &[])
+        .unwrap()
+        .l()
+        .unwrap();
+    let window_insets_types = env.find_class("android/view/WindowInsets$Type").unwrap();
+    let ime_type =
+        env.call_static_method(&window_insets_types, "ime", "()I", &[]).unwrap().i().unwrap();
+    let _ = if show {
+        env.call_method(&wic, "show", "(I)V", &[ime_type.into()])
+    } else {
+        env.call_method(&wic, "hide", "(I)V", &[ime_type.into()])
+    };
```

Comment:
Would be nice to have some docs on why it's done like that, also, do we really need that many unwraps here?

Reply:
Sure, I'll both check into whether there is a more ergonomic API available, and document why it has to be done this way. I think that the `jni` APIs are meant to be used with the `?` operator, so making the function fallible may work.